htlcswitch/quiescer.go

```diff
@@ -211,5 +224,38 @@ func (q *quiescer) drive() error {
 
 	q.sent = true
 
+	// Since we just sent an Stfu, we may have a newly quiesced state.
+	// If so, we will try to resolve any outstanding StfuReqs.
+	q.tryResolveStfuReq()
```

Comment:
If we defer this call then we don't need to have it in `recvStfu` because we can rely on `drive` to handle request resolution, since we always call `drive` after `recvStfu` anyway.

Reply:
I'm so conflicted on this. The clever part of me really likes this. The part of me that is concerned about how this code evolves over time is nervous about adding invisible control flow for things other than deallocation. I'd love to get a second opinion here. @yyforyongyu

Reply:
Once I finally got the full workflow I also think it's smart. I gotta say the process was not pleasant and I believe it's due to the current architecture state of `htlcswitch`. But I dare to ask this non-blocking wishful question, can we make things more linear?

Reply:
Linear how?

Reply:
linear in the sense we know what the current state is and take actions based on it, not sth like `tryResolveStfuReq` which requires some circuitous checking. Best illustrated in [code](https://github.com/yyforyongyu/lnd/commit/fd353ae747970d180565bb1aea106db3dc967f54).